### ISSUE TITLE
Problem with inline buttonsContainer in xhtml

### ISSUE THREAD

**Issue by skyynet** [heiseonline/shariff#264]
Probably I'm too stupid to set it up but since I always wanted to get rid of the xhtml errors caused by the data-services attribute of Shariff I tried to replace it inline with buttonsContainer:

```
<html>
<head>
<script language="JavaScript" type="text/javascript">
var buttonsContainer = $(".shariff");
new Shariff(buttonsContainer, {
    backendUrl: "/shariff/",
    services: "[&quot;facebook&quot;,&quot;twitter&quot;,&quot;xing&quot;]",
    orientation: "vertical",
    lang: "en",
});
</script>
</head>
<body>
<div class="shariff"></div>
<script type="text/javascript" src="/js/shariff.min.js"></script>
</body>
</html>
```

Unfortunately the defined options are not applied to the div. Ideas where's my error in reasoning? Thanks.

**Comment by liayn** [comment]
Your services definition is wrong. Is has to be an array of strings not a string.

**Comment by skyynet** [comment]
Thanks, @liayn 
The code I used is exactly the one that's given at [https://github.com/heiseonline/shariff](url) under "Constructor arguments" so I don't know what I've got to change to make it work. Further details highly appreciated :-)

**Comment by skyynet** [comment]
Thanks for your hint. Unfortunately I can't get it to work this way.

I set up two minimized test pages:
[old inline](https://skyynet.de/test)
[new buttonsContainer](https://skyynet.de/test_bc)

If you see something I do wrong, please give me a hint.

**Comment by liayn** [comment]
Addendum: You again messed up the `services` definition: Compare my code above and yours.

**Comment by skyynet** [comment]
Thanks, @liayn 
I messed up the `services` definition while trying to fix the trailing quote :-)
Anyways, the hint that led me to the correct code was the one about the inline code being executed before the DOM is ready to use.
I placed some JS at the end of the pages as Google Page Speed always bugs me about JS preventing to load the page faster otherwise.

I now use this code and everything works fine on my [homepage](https://skyynet.de). Thus I removed the test scripts mentioned before in this conversation.
`$(window).load(function() {
 var buttonsContainer = $(".shariff");                    
 new Shariff(buttonsContainer, {
    backendUrl: "/shariff/",
    services: ['facebook', 'twitter', 'xing'],
    orientation: "vertical",
    lang: "de",
 });
}); `
